extensions/ext/GLSL_EXT_spirv_intrinsics.txt

```diff
@@ -0,0 +1,1020 @@
+            constant-expression , spirv_decorate_id-parameter-list
+
+        spirv_decorate_string-parameter-list:
+
+            literal-string
+            literal-string , spirv_decorate_string-parameter-list
+
+        The `spirv_decorate` qualifier inserts an *OpDecorate* or
+        *OpMemberDecorate* instruction into the generated SPIR-V code,
+        taking the first integer parameter as its `Decoration` operand,
+        and further parameters as `Extra Operands`.
+        If the `spirv_decorate` qualifier is applied to a struct member,
+        *OpMemberDecorate* is inserted, otherwise *OpDecorate* is.
+        Integer and floating point values used must be expressible in a
+        32-bit range.
+        An error will be generated if the front-end compiler detects a
+        value outside of the 32-bit range.
+
+        The `spirv_decorate_id` qualifier inserts an *OpDecorateId*
+        instruction into the generated SPIR-V code, taking the first
+        integer parameter as its `Decoration` operand, and further
+        parameters as `Extra Operands`.
+        Parameters to this qualifier must be constant expressions as defined
+        in section 4.3.3, Constant Expressions.
+
+        The `spirv_decorate_string` qualifier inserts an *OpDecorateString*
+        or *OpMemberDecorateString* instruction into the generated SPIR-V
+        code, taking the first integer parameter as its `Decoration`
+        operand, the first string as the first `Literal` string argument,
+        and string parameters as `Optional Literals`.
+        If the `spirv_decorate_string` qualifier is applied to a struct
+        member, *OpMemberDecorateString* is inserted, otherwise
+        *OpDecorateString* is.
+
+        Any extensions or capabilities needed by the decoration must be
+        declared using SPIR-V Requirements Declarations (see section 3.9).
+
+        Multiple `spirv_decoration` qualifiers can be added to a single
+        declaration or expression.
+
+    Add a new section 4.14. SPIR-V Types
+
+        SPIR-V types not known to or exposed by the GLSL compiler can
+        be added by use of the `spirv_type` specifier.
+
+        This functionality is only available when generating SPIR-V.
+
+        The `spirv_type()` specifier can be used in place of a type name
+        for variable declarations.
+
+        spirv_type-specifier:
+
+            `spirv_type` ( spirv_instruction-qualifier-list )
+            `spirv_type` ( spirv_instruction-qualifier-list, spirv_type-parameter-list )
+            `spirv_type` ( spirv-requirements-list, spirv_instruction-qualifier-list )
+            `spirv_type` ( spirv-requirements-list, spirv_instruction-qualifier-list, spirv_type-parameter-list )
+
+        spirv_type-parameter-list:
+
+            spirv_type-parameter
+            spirv_type-parameter , spirv_type-parameter-list
+
+        spirv_type-parameter:
+
+            constant-expression
+            literal
```

Comment:
You don't need this because a literal is a type of constant expression.

Reply:
```suggestion
```

Fair enough